### PR TITLE
refactor: new collection panel UI for mobile 

### PR DESCRIPTION
**Pull request by JoelJacobStephen**
### Changes

- Redesigned the collection panel for mobile where it slides into view and occupies the full height of the screen, instead of using a 3rd pane to display the collections. 

### Fixes

- Fixed uneven width of send button in the REST page on mobile devices.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### CODE REVIEW

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue [hunk 1]
```diff
@@ -30,25 +30,40 @@
...
+      v-if="!mdAndLarger && SIDEBAR && hasSidebar"
+      :show="show"
+      :title="t('tab.collections')"
+      @close="SIDEBAR = !SIDEBAR"
+    >
+      <template #content>
+        <div class="!overflow-auto">
+          <slot name="sidebar" />
+        </div>
+      </template>
```
Comment (anwarulislam) [inline packages/hoppscotch-common/src/components/app/PaneLayout.vue:52]: Change the name #content to something meaningful as slideovercontent

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue [hunk 2]
```diff
@@ -30,25 +30,40 @@
...
       class="flex flex-col !overflow-auto bg-primaryContrast"
     >
       <slot name="sidebar" />
     </Pane>
+
+    <!-- Mobile Collections Layout -->
+    <SmartSlideOver
+      v-if="!mdAndLarger && SIDEBAR && hasSidebar"
+      :show="show"
+      :title="t('tab.collections')"
```
Comment (anwarulislam) [inline packages/hoppscotch-common/src/components/app/PaneLayout.vue:45]: Pass the title as a prop. If we want to use SlideOver later from a different page/component we can do that.

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue [hunk 3]
```diff
@@ -70,6 +85,8 @@ const props = defineProps({
   },
 })
 
+const show = computed(() => !!(SIDEBAR && hasSidebar.value))
```
Comment (anwarulislam) [inline packages/hoppscotch-common/src/components/app/PaneLayout.vue:88]: Change variable name to `showSlideOver`

**Review by anwarulislam — CHANGES_REQUESTED** [review]

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue (hunk 1 above)
Comment (JoelJacobStephen) [reply packages/hoppscotch-common/src/components/app/PaneLayout.vue]: Renamed #content to #slideovercontent

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue (hunk 3 above)
Comment (JoelJacobStephen) [reply packages/hoppscotch-common/src/components/app/PaneLayout.vue]: Renamed show variable to showSlideOver

File: packages/hoppscotch-common/src/components/app/PaneLayout.vue (hunk 2 above)
Comment (JoelJacobStephen) [reply packages/hoppscotch-common/src/components/app/PaneLayout.vue:45]: Title is being passed as a prop to SlideOver Component.

**Comment by AndrewBastin** [comment]
Closing this PR due to irrelevancy after the decision to drop support mobile UI

**Pull request closed without merging**